controllers/snapshot/snapshot_adapter.go

```diff
@@ -306,6 +307,27 @@ func (a *Adapter) EnsureCreationOfEphemeralEnvironments() (controller.OperationR
 					"integrationTestScenario.Name", integrationTestScenario.Name)
 				return controller.StopProcessing()
 			}
+			if clienterrors.IsInvalid(err) {
+
+				//a.logger.Error(err, "environment or deploymentTargetClass referenced by integrationTestScenario is invalid", "integrationTestScenario.Name", integrationTestScenario.Name)
+				if !gitops.IsSnapshotMarkedAsFailed(a.snapshot) {
+					var (
+						errKind   string = "<unknown>"
+						errName   string = "<unknown>"
+						statusErr *clienterrors.StatusError
+					)
+					if ok := errors.As(err, &statusErr); ok {
+						errKind = statusErr.Status().Details.Kind
+						errName = statusErr.Status().Details.Name
+					}
+					a.logger.Error(err, fmt.Sprintf("Resource %s (%s) referenced by integrationTestScenario is invalid", errName, errKind), "integrationTestScenario.Name", integrationTestScenario.Name)
+					_, err = gitops.MarkSnapshotAsFailed(a.client, a.context, a.snapshot, fmt.Sprintf("Object %s (%s) associated with snapshot %s is invalid: %s.", errName, errKind, a.snapshot.Spec.Application, err))
```

Comment:
It might be ```suggestion
					_, err = gitops.MarkSnapshotAsFailed(a.client, a.context, a.snapshot, fmt.Sprintf("Object %s (%s) associated with snapshot %s is invalid: %s.", errName, errKind, a.snapshot.Name, err))
``` because it says `snapshot %s`

Reply:
```suggestion
					_, err = gitops.MarkSnapshotAsFailed(a.client, a.context, a.snapshot, fmt.Sprintf("Resource %s (%s) associated with snapshot %s is invalid: %s.", errName, errKind, a.snapshot.Spec.Application, err))
```

Reply:
Updated to reflect resource & a more clear error message that refers to the ITS as opposed to the snapshot [here](https://github.com/redhat-appstudio/integration-service/commit/d03da3f685c1c4b4d3e036780b3a2401a384bf03#diff-8e718c6bf06f754db7e9e94c6f74b5c6704941dff95d637289e3c4cc194c95a0R316)